### PR TITLE
Added controller for private runners of Github Actions

### PR DESCRIPTION
**Pull request by RustamGimadiev**

### CODE REVIEW

File: modules/cicd/github-actions/variables.tf [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+variable "tags" {
+  type        = map(string)
+  default     = {}
+  description = "A tags for attaching to new created AWS resources"
+}
+
+variable "github_token" {
+  type        = string
+  description = "A GitHub token for application"
+  default     = null
```
Comment (majeX) [inline modules/cicd/github-actions/variables.tf:52]: `sensitive = true` ?

File: modules/cicd/github-actions/main.tf [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    }
+  })
+  filename = "${path.root}/${var.argocd.path}/ns-${local.namespace}.yaml"
+}
+
+resource "local_file" "github_runners" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
```
Comment (majeX) [inline modules/cicd/github-actions/main.tf:33]: Terraform now supports `depends_on` for modules, is this still necessary?

File: modules/cicd/github-actions/main.tf [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  name          = local.name
+  repository    = local.repository
+  chart         = local.chart
+  version       = var.chart_version
+  namespace     = local.namespace
+  recreate_pods = true
+  timeout       = 1200
+
+  dynamic "set" {
+    for_each = merge(local.conf)
```
Comment (majeX) [inline modules/cicd/github-actions/main.tf:101]: Is this a hack I'm not familiar with? Merge what with what?

File: modules/cicd/github-actions/main.tf (hunk 2 above)
Comment (RustamGimadiev) [reply modules/cicd/github-actions/main.tf:33]: let's leave it as it is, we can pass an empty list as dependencies here, just for compatibility with terraform <12 and others modules

**Review by majeX — APPROVED** [review]
👌

File: modules/cicd/github-actions/variables.tf [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  type        = map(string)
+  description = "describe your variable"
+  default     = {}
+}
+
+variable "runner_deployment_spec" {
+  type        = any
+  description = "A custom Github Runner configuration"
+  default = {
+    "replicas" = 2
```
Comment (akastav) [inline modules/cicd/github-actions/variables.tf:66]: Why 2 by default?

Comment (viatcheslavmogilevsky) [reply modules/cicd/github-actions/variables.tf]: Maybe to catch possible issues with multiple runners - those issues won't happened when we have single runner

**Review by akastav — APPROVED** [review]

**Pull request merged**